
<!doctype html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      
      
      
      
        <link rel="prev" href="../class-applications-of-evo/">
      
      
      
      <link rel="icon" href="/images/favicon.ico">
      <meta name="generator" content="mkdocs-1.5.3, mkdocs-material-9.5.14">
    
    
      
        <title>Human evolution - Fundamentals of Evolution</title>
      
    
    
      <link rel="stylesheet" href="../assets/stylesheets/main.10ba22f1.min.css">
      
        
        <link rel="stylesheet" href="../assets/stylesheets/palette.06af60db.min.css">
      
      


    
    
      
    
    
      
        
        
        <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,300i,400,400i,700,700i%7CRoboto+Mono:400,400i,700,700i&display=fallback">
        <style>:root{--md-text-font:"Roboto";--md-code-font:"Roboto Mono"}</style>
      
    
    
      <link rel="stylesheet" href="../stylesheets/extra.css">
    
    <script>__md_scope=new URL("..",location),__md_hash=e=>[...e].reduce((e,_)=>(e<<5)-e+_.charCodeAt(0),0),__md_get=(e,_=localStorage,t=__md_scope)=>JSON.parse(_.getItem(t.pathname+"."+e)),__md_set=(e,_,t=localStorage,a=__md_scope)=>{try{t.setItem(a.pathname+"."+e,JSON.stringify(_))}catch(e){}}</script>
    
      

    
    
    
  </head>
  
  
    
    
      
    
    
    
    
    <body dir="ltr" data-md-color-scheme="slate" data-md-color-primary="light-blue" data-md-color-accent="indigo">
  
    
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="__drawer" autocomplete="off">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="__search" autocomplete="off">
    <label class="md-overlay" for="__drawer"></label>
    <div data-md-component="skip">
      
        
        <a href="#human-evolution" class="md-skip">
          Skip to content
        </a>
      
    </div>
    <div data-md-component="announce">
      
    </div>
    
    
      

  

<header class="md-header md-header--shadow" data-md-component="header">
  <nav class="md-header__inner md-grid" aria-label="Header">
    <a href=".." title="Fundamentals of Evolution" class="md-header__button md-logo" aria-label="Fundamentals of Evolution" data-md-component="logo">
      
  
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 8a3 3 0 0 0 3-3 3 3 0 0 0-3-3 3 3 0 0 0-3 3 3 3 0 0 0 3 3m0 3.54C9.64 9.35 6.5 8 3 8v11c3.5 0 6.64 1.35 9 3.54 2.36-2.19 5.5-3.54 9-3.54V8c-3.5 0-6.64 1.35-9 3.54Z"/></svg>

    </a>
    <label class="md-header__button md-icon" for="__drawer">
      
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M3 6h18v2H3V6m0 5h18v2H3v-2m0 5h18v2H3v-2Z"/></svg>
    </label>
    <div class="md-header__title" data-md-component="header-title">
      <div class="md-header__ellipsis">
        <div class="md-header__topic">
          <span class="md-ellipsis">
            Fundamentals of Evolution
          </span>
        </div>
        <div class="md-header__topic" data-md-component="header-topic">
          <span class="md-ellipsis">
            
              Human evolution
            
          </span>
        </div>
      </div>
    </div>
    
      
        <form class="md-header__option" data-md-component="palette">
  
    
    
    
    <input class="md-option" data-md-color-media="" data-md-color-scheme="slate" data-md-color-primary="light-blue" data-md-color-accent="indigo"  aria-label="Switch to light mode"  type="radio" name="__palette" id="__palette_0">
    
      <label class="md-header__button md-icon" title="Switch to light mode" for="__palette_1" hidden>
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 2a7 7 0 0 0-7 7c0 2.38 1.19 4.47 3 5.74V17a1 1 0 0 0 1 1h6a1 1 0 0 0 1-1v-2.26c1.81-1.27 3-3.36 3-5.74a7 7 0 0 0-7-7M9 21a1 1 0 0 0 1 1h4a1 1 0 0 0 1-1v-1H9v1Z"/></svg>
      </label>
    
  
    
    
    
    <input class="md-option" data-md-color-media="" data-md-color-scheme="default" data-md-color-primary="indigo" data-md-color-accent="indigo"  aria-label="Switch to dark mode"  type="radio" name="__palette" id="__palette_1">
    
      <label class="md-header__button md-icon" title="Switch to dark mode" for="__palette_0" hidden>
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 2a7 7 0 0 1 7 7c0 2.38-1.19 4.47-3 5.74V17a1 1 0 0 1-1 1H9a1 1 0 0 1-1-1v-2.26C6.19 13.47 5 11.38 5 9a7 7 0 0 1 7-7M9 21v-1h6v1a1 1 0 0 1-1 1h-4a1 1 0 0 1-1-1m3-17a5 5 0 0 0-5 5c0 2.05 1.23 3.81 3 4.58V16h4v-2.42c1.77-.77 3-2.53 3-4.58a5 5 0 0 0-5-5Z"/></svg>
      </label>
    
  
</form>
      
    
    
      <script>var media,input,key,value,palette=__md_get("__palette");if(palette&&palette.color){"(prefers-color-scheme)"===palette.color.media&&(media=matchMedia("(prefers-color-scheme: light)"),input=document.querySelector(media.matches?"[data-md-color-media='(prefers-color-scheme: light)']":"[data-md-color-media='(prefers-color-scheme: dark)']"),palette.color.media=input.getAttribute("data-md-color-media"),palette.color.scheme=input.getAttribute("data-md-color-scheme"),palette.color.primary=input.getAttribute("data-md-color-primary"),palette.color.accent=input.getAttribute("data-md-color-accent"));for([key,value]of Object.entries(palette.color))document.body.setAttribute("data-md-color-"+key,value)}</script>
    
    
    
      <label class="md-header__button md-icon" for="__search">
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5Z"/></svg>
      </label>
      <div class="md-search" data-md-component="search" role="dialog">
  <label class="md-search__overlay" for="__search"></label>
  <div class="md-search__inner" role="search">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" aria-label="Search" placeholder="Search" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="search-query" required>
      <label class="md-search__icon md-icon" for="__search">
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5Z"/></svg>
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11h12Z"/></svg>
      </label>
      <nav class="md-search__options" aria-label="Search">
        
        <button type="reset" class="md-search__icon md-icon" title="Clear" aria-label="Clear" tabindex="-1">
          
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12 19 6.41Z"/></svg>
        </button>
      </nav>
      
        <div class="md-search__suggest" data-md-component="search-suggest"></div>
      
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" data-md-scrollfix>
        <div class="md-search-result" data-md-component="search-result">
          <div class="md-search-result__meta">
            Initializing search
          </div>
          <ol class="md-search-result__list" role="presentation"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
    
    
      <div class="md-header__source">
        <a href="https://github.com/eaton-lab/fundamentals" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 6.5.1 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2023 Fonticons, Inc.--><path d="M439.55 236.05 244 40.45a28.87 28.87 0 0 0-40.81 0l-40.66 40.63 51.52 51.52c27.06-9.14 52.68 16.77 43.39 43.68l49.66 49.66c34.23-11.8 61.18 31 35.47 56.69-26.49 26.49-70.21-2.87-56-37.34L240.22 199v121.85c25.3 12.54 22.26 41.85 9.08 55a34.34 34.34 0 0 1-48.55 0c-17.57-17.6-11.07-46.91 11.25-56v-123c-20.8-8.51-24.6-30.74-18.64-45L142.57 101 8.45 235.14a28.86 28.86 0 0 0 0 40.81l195.61 195.6a28.86 28.86 0 0 0 40.8 0l194.69-194.69a28.86 28.86 0 0 0 0-40.81z"/></svg>
  </div>
  <div class="md-source__repository">
    GitHub
  </div>
</a>
      </div>
    
  </nav>
  
</header>
    
    <div class="md-container" data-md-component="container">
      
      
        
          
        
      
      <main class="md-main" data-md-component="main">
        <div class="md-main__inner md-grid">
          
            
              
              <div class="md-sidebar md-sidebar--primary" data-md-component="sidebar" data-md-type="navigation" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    



<nav class="md-nav md-nav--primary" aria-label="Navigation" data-md-level="0">
  <label class="md-nav__title" for="__drawer">
    <a href=".." title="Fundamentals of Evolution" class="md-nav__button md-logo" aria-label="Fundamentals of Evolution" data-md-component="logo">
      
  
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 8a3 3 0 0 0 3-3 3 3 0 0 0-3-3 3 3 0 0 0-3 3 3 3 0 0 0 3 3m0 3.54C9.64 9.35 6.5 8 3 8v11c3.5 0 6.64 1.35 9 3.54 2.36-2.19 5.5-3.54 9-3.54V8c-3.5 0-6.64 1.35-9 3.54Z"/></svg>

    </a>
    Fundamentals of Evolution
  </label>
  
    <div class="md-nav__source">
      <a href="https://github.com/eaton-lab/fundamentals" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 6.5.1 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2023 Fonticons, Inc.--><path d="M439.55 236.05 244 40.45a28.87 28.87 0 0 0-40.81 0l-40.66 40.63 51.52 51.52c27.06-9.14 52.68 16.77 43.39 43.68l49.66 49.66c34.23-11.8 61.18 31 35.47 56.69-26.49 26.49-70.21-2.87-56-37.34L240.22 199v121.85c25.3 12.54 22.26 41.85 9.08 55a34.34 34.34 0 0 1-48.55 0c-17.57-17.6-11.07-46.91 11.25-56v-123c-20.8-8.51-24.6-30.74-18.64-45L142.57 101 8.45 235.14a28.86 28.86 0 0 0 0 40.81l195.61 195.6a28.86 28.86 0 0 0 40.8 0l194.69-194.69a28.86 28.86 0 0 0 0-40.81z"/></svg>
  </div>
  <div class="md-source__repository">
    GitHub
  </div>
</a>
    </div>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href=".." class="md-nav__link">
        
  
  <span class="md-ellipsis">
    About/Ethos
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="../syllabus/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Syllabus
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="../schedule/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Schedule
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
  
  
    
    
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_4" >
        
          
          <label class="md-nav__link" for="__nav_4" id="__nav_4_label" tabindex="0">
            
  
  <span class="md-ellipsis">
    Module 1 - Adaptation
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_4_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_4">
            <span class="md-nav__icon md-icon"></span>
            Module 1 - Adaptation
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../class-evol-theory/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Theory of Evolution
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../class-tree-thinking/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Tree Thinking
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../class-biodiversity/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Biodiversity
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../class-adaptation/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Adaptation
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../class-genetical-theory/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Genetical Theory
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
  
  
    
    
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_5" >
        
          
          <label class="md-nav__link" for="__nav_5" id="__nav_5_label" tabindex="0">
            
  
  <span class="md-ellipsis">
    Module 2 - Genetic variation
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_5_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_5">
            <span class="md-nav__icon md-icon"></span>
            Module 2 - Genetic variation
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../class-genetic-variation/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Genetic Variation
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../class-modeling-evolution/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Modeling Evolution
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../class-ancestry-coalescent/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Ancestry
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../class-population-structure/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Population Structure
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../class-phylogenomics/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Phylogenomics
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../class-2-spatial/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Spatial genetics
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../class-2-speciation/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Speciation
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
  
  
    
    
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_6" >
        
          
          <label class="md-nav__link" for="__nav_6" id="__nav_6_label" tabindex="0">
            
  
  <span class="md-ellipsis">
    Module 3 - Pattern/Process
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_6_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_6">
            <span class="md-nav__icon md-icon"></span>
            Module 3 - Pattern/Process
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../class-3-genomes-1/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Genomes I
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../class-3-genomes-2/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Genomes II
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../class-3-development/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Development
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../class-3-macroevolution/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Macroevolution
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../class-3-synthesis/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Synthesis
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
    
  
  
  
    
    
    
    
    <li class="md-nav__item md-nav__item--active md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_7" checked>
        
          
          <label class="md-nav__link" for="__nav_7" id="__nav_7_label" tabindex="0">
            
  
  <span class="md-ellipsis">
    Module 4 - Applications
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_7_label" aria-expanded="true">
          <label class="md-nav__title" for="__nav_7">
            <span class="md-nav__icon md-icon"></span>
            Module 4 - Applications
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../class-applications-of-evo/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Applications of Evol Theory
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
    
  
  
  
    <li class="md-nav__item md-nav__item--active">
      
      <input class="md-nav__toggle md-toggle" type="checkbox" id="__toc">
      
      
        
      
      
        <label class="md-nav__link md-nav__link--active" for="__toc">
          
  
  <span class="md-ellipsis">
    Human evolution
  </span>
  

          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <a href="./" class="md-nav__link md-nav__link--active">
        
  
  <span class="md-ellipsis">
    Human evolution
  </span>
  

      </a>
      
        

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#outline" class="md-nav__link">
    <span class="md-ellipsis">
      Outline
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Outline">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#assigned-reading" class="md-nav__link">
    <span class="md-ellipsis">
      Assigned reading
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#slides" class="md-nav__link">
    <span class="md-ellipsis">
      Slides
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#references" class="md-nav__link">
    <span class="md-ellipsis">
      References
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#lecture-notes" class="md-nav__link">
    <span class="md-ellipsis">
      Lecture Notes
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Lecture Notes">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#human-evolution_1" class="md-nav__link">
    <span class="md-ellipsis">
      Human Evolution
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#fossil-evidence" class="md-nav__link">
    <span class="md-ellipsis">
      Fossil evidence
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#eugenics-history" class="md-nav__link">
    <span class="md-ellipsis">
      Eugenics history
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#how-did-humans-evolve" class="md-nav__link">
    <span class="md-ellipsis">
      How did humans evolve?
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#ancestors-out-of-africa" class="md-nav__link">
    <span class="md-ellipsis">
      Ancestors out of Africa
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#human-genetics" class="md-nav__link">
    <span class="md-ellipsis">
      Human genetics
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#why-study-human-genetics" class="md-nav__link">
    <span class="md-ellipsis">
      Why study human genetics?
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#ancient-dna" class="md-nav__link">
    <span class="md-ellipsis">
      Ancient DNA
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#human-genetics-history" class="md-nav__link">
    <span class="md-ellipsis">
      Human genetics history
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#variant-centric-view-of-human-genetic-variation" class="md-nav__link">
    <span class="md-ellipsis">
      Variant-centric view of human genetic variation
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#missing-heritability" class="md-nav__link">
    <span class="md-ellipsis">
      Missing heritability
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#gwas" class="md-nav__link">
    <span class="md-ellipsis">
      GWAS
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#polygenic-score" class="md-nav__link">
    <span class="md-ellipsis">
      Polygenic score
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#controversial-gwa-studies" class="md-nav__link">
    <span class="md-ellipsis">
      Controversial GWA studies
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
    </ul>
  
</nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="https://docs.google.com/presentation/d/1VYaX-srzQg2l1Xw4ommRIsy0b4olgZPYT2LD4_18v-k/edit?usp=sharing" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    review-slides
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              
              <div class="md-sidebar md-sidebar--secondary" data-md-component="sidebar" data-md-type="toc" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#outline" class="md-nav__link">
    <span class="md-ellipsis">
      Outline
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Outline">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#assigned-reading" class="md-nav__link">
    <span class="md-ellipsis">
      Assigned reading
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#slides" class="md-nav__link">
    <span class="md-ellipsis">
      Slides
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#references" class="md-nav__link">
    <span class="md-ellipsis">
      References
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#lecture-notes" class="md-nav__link">
    <span class="md-ellipsis">
      Lecture Notes
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Lecture Notes">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#human-evolution_1" class="md-nav__link">
    <span class="md-ellipsis">
      Human Evolution
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#fossil-evidence" class="md-nav__link">
    <span class="md-ellipsis">
      Fossil evidence
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#eugenics-history" class="md-nav__link">
    <span class="md-ellipsis">
      Eugenics history
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#how-did-humans-evolve" class="md-nav__link">
    <span class="md-ellipsis">
      How did humans evolve?
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#ancestors-out-of-africa" class="md-nav__link">
    <span class="md-ellipsis">
      Ancestors out of Africa
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#human-genetics" class="md-nav__link">
    <span class="md-ellipsis">
      Human genetics
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#why-study-human-genetics" class="md-nav__link">
    <span class="md-ellipsis">
      Why study human genetics?
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#ancient-dna" class="md-nav__link">
    <span class="md-ellipsis">
      Ancient DNA
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#human-genetics-history" class="md-nav__link">
    <span class="md-ellipsis">
      Human genetics history
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#variant-centric-view-of-human-genetic-variation" class="md-nav__link">
    <span class="md-ellipsis">
      Variant-centric view of human genetic variation
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#missing-heritability" class="md-nav__link">
    <span class="md-ellipsis">
      Missing heritability
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#gwas" class="md-nav__link">
    <span class="md-ellipsis">
      GWAS
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#polygenic-score" class="md-nav__link">
    <span class="md-ellipsis">
      Polygenic score
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#controversial-gwa-studies" class="md-nav__link">
    <span class="md-ellipsis">
      Controversial GWA studies
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
    </ul>
  
</nav>
                  </div>
                </div>
              </div>
            
          
          
            <div class="md-content" data-md-component="content">
              <article class="md-content__inner md-typeset">
                
                  

  
  


<h1 id="human-evolution">Human evolution<a class="headerlink" href="#human-evolution" title="Permanent link">&para;</a></h1>
<hr />
<h2 id="outline">Outline<a class="headerlink" href="#outline" title="Permanent link">&para;</a></h2>
<ul>
<li>Lecture: Human population genomics<ul>
<li>Fossils, Population Genetics, and current knowledge.</li>
<li>History, racism, and ethics.</li>
<li>Applications, Impacts, and Society.</li>
</ul>
</li>
<li><em>Break</em></li>
<li>Lecture: <ul>
<li>Crispr, germline editing.</li>
<li>...</li>
</ul>
</li>
</ul>
<hr />
<h3 id="assigned-reading">Assigned reading<a class="headerlink" href="#assigned-reading" title="Permanent link">&para;</a></h3>
<ul>
<li>Futuyma textbook chapter on Human Evolution</li>
</ul>
<!-- GCBias: [https://gcbias.org/2022/07/12/genetic-ancestry-groups-and-genetic-similarity/](https://gcbias.org/2022/07/12/genetic-ancestry-groups-and-genetic-similarity/)
- science.org/content/article/how-neanderthals-lost-their-y-chromosome
-->
<!-- who has access/rights to historical samples -->
<!-- who is member of a group? -->

<h2 id="slides">Slides<a class="headerlink" href="#slides" title="Permanent link">&para;</a></h2>
<p><a href="https://docs.google.com/presentation/d/1qZ7V8a7RMXAPHhzv2eexH0XOdEvkuGRZh3cJvhXLX9A/edit?usp=sharing">https://docs.google.com/presentation/d/1qZ7V8a7RMXAPHhzv2eexH0XOdEvkuGRZh3cJvhXLX9A/edit?usp=sharing</a></p>
<hr />
<h2 id="references">References<a class="headerlink" href="#references" title="Permanent link">&para;</a></h2>
<ul>
<li>https://en.wikipedia.org/wiki/The_Descent_of_Man,_and_Selection_in_Relation_to_Sex</li>
<li>https://en.wikipedia.org/wiki/Eugenics#Origin_and_development</li>
<li>https://en.wikipedia.org/wiki/Missing_heritability_problem</li>
<li>https://en.wikipedia.org/wiki/Polygenic_score</li>
<li>...</li>
</ul>
<hr />
<h2 id="lecture-notes">Lecture Notes<a class="headerlink" href="#lecture-notes" title="Permanent link">&para;</a></h2>
<h3 id="human-evolution_1">Human Evolution<a class="headerlink" href="#human-evolution_1" title="Permanent link">&para;</a></h3>
<p>Human's evolved and continue to evolve just like all other organisms on Earth.
(Fig. 21.1).</p>
<p>The common ancestry of human's with other primates and all other organisms 
has been hypothesized many times independently. It was famously described 
and supported by abundant evidence by Darwin in both the 'Origin of Species'
and 'The Descent of Man'. (The Descent of Man is a clever title). By the time
Darwin published the Descent of Man (1871) the idea of Human evolution was
broadly discussed. Darwin presented evidence from morphology, embryology, and
behavior. </p>
<h3 id="fossil-evidence">Fossil evidence<a class="headerlink" href="#fossil-evidence" title="Permanent link">&para;</a></h3>
<p>The strongest evidence initially came from morphological comparisons. We share
many similar features with our closest relatives. The estimated divergence time
between humans and the lineage that includes Chimpanzees and Bonobos is 
approximately 7 Mya. Since that time, both lineages have likely undergone 
significant divergence and adaptations from their common ancestor. </p>
<p>Evolution of the human lineage was not a single directional change, but rather,
included many sublineages that diverged from each other over time, dispersed
geographically, and independently adapted to different conditions. All of these
lineages except for our own eventually went extinct.</p>
<p>The evidence for this complex history of the human lineage is found in fossil
remains mostly throughout Africa and Eurasia. </p>
<h3 id="eugenics-history">Eugenics history<a class="headerlink" href="#eugenics-history" title="Permanent link">&para;</a></h3>
<p>Darwin's Descent book refers to many human populations as "savages" in 
discussing the evolution of behavior and societies, using language that is 
offensive today. </p>
<p>The idea of 'social Darwinism' -- the effects of natural or artifical 
selection on human societies -- had been widely discussed since the Origin 
of Species. Darwin wrote on how modern/Victorian society was largely removed
from the type of natural selection that acts on other societies that 
still interact more intensely with nature. </p>
<p>The idea of organized action to select on human population, or 
<em>eugenics</em>, was largely developed and first championed by Francis Galton, 
a half-cousin of Darwin. Darwin strongly disagreed with Galton's theory. The
first usage of the term eugenics was in 1883, one year after Darwin's death.</p>
<p>Eugenics became an academic discipline, especially in Britain and the US. 
Interestingly, many of the ideas of eugenics rely more strongly on the idea
of <em>genetic determinism</em> -- simply that human traits are genetically 
determined -- than they do to Darwin's theory of evolution (i.e., these ideas
do not relate to adaptation or common ancestry of species, but simply to 
the idea of genetic inheritance. In fact they would apply equally to a 
LaMarkian view of evolution). 
Many early geneticists were not Darwinians, and evolutionary 
theory was not needed for eugenic policies based on genetic determinism. </p>
<p>In fact, eugenic practices have been well documented in human history dating
back long before the development of evolutionary theory. Given that human 
populations developed selective breeding methods for plants and animals, it
is not that surprising that societies could have applied such practices to 
humans. This includes documentation of infanticide against children born
with physical abnormalities. Of stories about selective breeding for warrior
or athletic ability in Greek and Samnite culters. These later examples are 
consideredc likely to be stories as opposed to widespread societal 
practices.</p>
<p>The British and American Eugenics societies were formed in 1907 and 1921, 
respectively. The academic nature of these writings was used as justification
for many social and governmental practices of xenophobia. Eugenics justified
"scientific racism". </p>
<p>Eugenic policies in the US were implemented in the early 1900s that promoted
the sterilization of people deemed to have mental disorders, to justify 
racist practices including segregation and laws against interracial marriage
(anti-miscegenation laws), 
and to justify restrictive and racist immigration policies. Many of the most
famous politicians of this era supported eugenics, including Winston Churchill,
Teddy Roosevelt. There were many outspoken critics of the eugenics movement
as well, such as the Columbia University anthropologist Franz Boas, and to
some extent, the Catholic church.</p>
<p>The scientific reputation of eugenics declined after the second world war
largely in response to the racial policies enacted by Nazi Germany. By the 
end of the war Germany had committed mass murder of people that were 
considered to be "unfit" or "degenerate". In addition the approximately 
6 million Jews murdered in the holocaust, this included mass killings of 
disabled people, homosexuals, and other minority groups. </p>
<p>Race laws and practices in the US were explicitly used as models by 
Nazi Germany in developing the 
Nuremberg Laws which stripped Jews of their citizenship rights. In some 
ways, the racial laws in Germany were less severe than those in the US at 
this time. American law during the 20<sup>th</sup> century asserted that a person with
even one ancestor of black ancestry ("one drop rule") is considered black. 
This law was used to subjugate people. The Nazi's considered this too extreme
to apply to Jewish ancestry.</p>
<!-- Based on late 20th-century DNA analysis and a preponderance of historical evidence, US president Thomas Jefferson is widely believed to have fathered the six mixed-race children with his slave Sally Hemings, who was herself three-quarters white and a paternal half-sister of his wife Martha Wayles Jefferson.[quote 1] Four of them survived to adulthood.[7] Under Virginia law of the time, while their seven-eighths European ancestry would have made them legally white if they’d been free, being born to an enslaved mother made them automatically enslaved from birth. Jefferson allowed the two oldest to escape in 1822 (freeing them legally was a public action he elected to avoid because he would have had to gain permission from the state legislature); the two youngest he freed in his 1826 will. Three of the four entered white society as adults. Subsequently, their descendants identified as white.  -->

<p>Eugenic practices have been implemented even in recent decades, and are
enacted in many countries to this day in usually more subtle ways. For 
example, as recently as 1984 Singapore provided financial incentives to 
highly educated women to encourage them to have more children. The American
comedic film "Idiocracy" (2006) by Mike Judge presents a story line in which
a man undergoes hibernation and wakes up in the future to find that people
with less education reproduced more, such that society had become less 
intelligent. This is a common eugenicist trope, which assumes that intelligence
is heritable, and that some organized practice should encourage some to 
reproduce more than others. James Watson...</p>
<p>Prenatal screening -- examining embryonic fluid for genetic evidence of 
abnormalities -- is considered by some to be a form of eugenics as it may 
lead to the abortion of fetuses with undesirable traits. The prospect of 
human germline editing with CRISPR faces that same criticisms. Even 
environmental arguments about limiting population size -- especially arguments
which lay blame on the most populous regions of the globe, like India and 
China -- may be considered eugenicist and racist. Should 'natalist' arguments
-- arguments about how many children someone should have, or be allowed to 
have -- be part of our environmentalist rhetoric? </p>
<h3 id="how-did-humans-evolve">How did humans evolve?<a class="headerlink" href="#how-did-humans-evolve" title="Permanent link">&para;</a></h3>
<p>Humans differ from our closest relatives most clearly in brain size, 
in being bipedal, hairless, and in infant develoment. </p>
<h3 id="ancestors-out-of-africa">Ancestors out of Africa<a class="headerlink" href="#ancestors-out-of-africa" title="Permanent link">&para;</a></h3>
<h3 id="human-genetics">Human genetics<a class="headerlink" href="#human-genetics" title="Permanent link">&para;</a></h3>
<p>Bifurcating trees are a poor model for human genetic diversity. Throughout
our relatively young history, humans spread rapidly around the globe, giving
rise to highly structured populations that were non-randomly mating. However,
these populations were also not completely isolated. Frequent dispersal, 
interactions, trade, warfare, etc. brought human populations into contact 
recurrently throughout their history, such that human history represents more
of a network than a tree.</p>
<p>In addition, because the divergence time between human populations is so low, 
most genetic variation among populations represents ancestral variation that
has been sorted among populations, rather than differences deriving from new
mutations that have arisen within each individual group. The latter
does exist, the human population is accumulating new mutations all the time, 
however, these tend to exist at very low frequencies unless they were subject
to strong positive selection. </p>
<p>More commonly, as populations dispersed around the globe they carried with 
them a subset of genetic variation from their ancestors -- the most genetic
variation is found in Africa, where humans have persisted the longest. In 
each semi-isolated population this variation can become fixed/lost through
genetic drift or natural selection. Thus, even though a particular variant
(let's say the A allele at gene 1) becomes fixed in some population, that 
does not mean that variant is absent from other populations. Quite the 
contrary, allelic variation at this gene is likely to exist among the rest
of the human population (e.g., the A allele may exist at 1% or 10%, or 50% 
elsewhere, and perhaps vary among other populations in its frequency.) It 
is very rare in humans that a single population becomes fixed for an allele 
that is fixed for an alternative allele in every other population. This is
a result of the pattern of recent human ancestry.</p>
<p>If one assumes humans across the globe are a single randomly mating population, it would result in only a 5–15% average error when predicting the proportion of observed heterozygotes at a locus. (Novembre and Peter). The departure from random mating
due to population structure is measured by Fst: How much more variance in 
genetic variation can be explained by subgrouping populations versus treating
humans as one big population. Modern genomic datasets can dissect fine-scale
population structure from enormous amounts of data. Even though the variation
explained by that population structure is overall relatively small (5-15%).</p>
<h3 id="why-study-human-genetics">Why study human genetics?<a class="headerlink" href="#why-study-human-genetics" title="Permanent link">&para;</a></h3>
<p>By studying human population genetic diversity we can better understand the 
history of geography, linguistics, archaeology, and how human societies and
cultures developed. It is a tool for studying history. Knowledge of population
structure has allowed medical genetics to build more robust GWAS methods 
for measuring disease risk, to identify regions of the genome that have 
undergone selection/adaptation; and allowed people to learn their own 
personal ancestry, which can be a profound experience especially for many 
people whose cultural or ethnic history was forcibly erased.</p>
<h3 id="ancient-dna">Ancient DNA<a class="headerlink" href="#ancient-dna" title="Permanent link">&para;</a></h3>
<p>Although most human genetic studies examine variation among present day 
humans to try to reconstruct history, increasingly we also can use data
from 'ancient DNA', extracted from archaeological samples. For example, 
such studies have shown that levels of differentiation among European 
regions did not always increase through history (until the very recent), 
but instead, actually decreased during the late neolithic period and Bronze
age, when three major populations intersected. (Lazaridis 2014)</p>
<h3 id="human-genetics-history">Human genetics history<a class="headerlink" href="#human-genetics-history" title="Permanent link">&para;</a></h3>
<p>Cavalli-Sforza and colleagues ... </p>
<p>Lewontin 1972; "The Apportionment of Human Diversity" argued that variation
within populations is actually quite large relative to variation between
populations. This was only measurable for the first time using allozymes
(early genetic technique). This fundamentally changed the way many geneticists
thought about human genetic variation.</p>
<p>HapMap consortium (2005) .. genotyping arrays</p>
<p>Auton 2015: 1000 genomes project (1KGP) New York Genomes Center.</p>
<p>Simons Genome diversity project: 300 genomes from 142 diverse popualtions.</p>
<p>Fairley 2020. The international genome sample resource (2020).</p>
<h3 id="variant-centric-view-of-human-genetic-variation">Variant-centric view of human genetic variation<a class="headerlink" href="#variant-centric-view-of-human-genetic-variation" title="Permanent link">&para;</a></h3>
<p>A visual approach to see that most variants are shared among humans.</p>
<h3 id="missing-heritability">Missing heritability<a class="headerlink" href="#missing-heritability" title="Permanent link">&para;</a></h3>
<p>The 'missing heritability' problem is the fact that single genetic 
variation (SNVs) cannot account for much of the heritability of diseases,
behaviors, and other phenotypes. In other words, very few of the types of 
heritable variation among humans can be assigned to specific alleles with
known effect sizes. This is described as a significant problem for medicine,
since a persons susceptibility to a disease seems to be a combined effect
of many genes, as opposed to a single gene. </p>
<p>This problem was named in 2008, and was a disappointing outcome of the 
human genome project, an outcome of which was hoped to be a identification
of alleles causing many phenotypes. </p>
<p>As sequencing costs decreased researchers moved beyond examining variation in
single candidate genes, to instead examining variation across the entire 
genomes of many samples for correlations with phenotypes (e.g., disease) of
interest. Genome wide association studies (GWAS). </p>
<h3 id="gwas">GWAS<a class="headerlink" href="#gwas" title="Permanent link">&para;</a></h3>
<p>Human populations clearly exhibit morphological variation. This variation
is sometimes geographically structured, and some types of variation covary 
together. These correlated suites of traits, and their geographic structuring,
form the basis for historical categorizations of human populations into 
'races'. Linnaeus did not use the term "race" but instead described human
varieties, all as part of the same species. He described four varieties and
ranked them hierarchically. His Systema Naturae had huge impacts on 
thinking of human variation and race, and the "academic" study of human
variation as a form of natural history.</p>
<p>Many earlyu GWAS studies in the 2010s reported 'hits', finding candidate
genes for things like schizophrenia or intelligence, but these turned out
to include mostly false positives. They could not be replicated in future
studies. These 'candidate gene' studies in humans have been largely phased
out in favor of "polygenic risk scores".</p>
<p>This led to a dilemma. Standard genetics methods have long estimated large 
heritabilities such as 80% for traits such as height or intelligence, yet none
of the genes had been found despite sample sizes that, while small, should 
have been able to detect variants of reasonable effect size such as 1 inch or 
5 IQ points. If genes have such strong cumulative effects - where were they? 
Several resolutions have been proposed, that the missing heritability is 
some combination of: </p>
<ol>
<li>
<p>Environmental effects. Example, twin studies are highly biased b/c twins
are likely to be raised in similar environments, even when (rarely) separated.</p>
</li>
<li>
<p>Genetic effects may include many epigenetics.</p>
</li>
<li>
<p>Genetic effects are not actually additive (an assumption of these models).
Instead, they represente interactions among genes (epistasis).</p>
</li>
<li>
<p>The SNPs examined are not the cause of these diseases. Instead, it is rare
alleles that were not included in the SNP panel (WGS should help overcome this),
or they represent structural variation (newer methods are beginning to address
this, such as long read sequencing.)</p>
</li>
<li>
<p>Traits are mis-diagnosed or measured. Schizophenia may be multiple different
things as opposed to one disease. Intelligence is subjective.</p>
</li>
</ol>
<p>or other reasons.</p>
<p>https://www.newyorker.com/magazine/2021/09/13/can-progressives-be-convinced-that-genetics-matters</p>
<h3 id="polygenic-score">Polygenic score<a class="headerlink" href="#polygenic-score" title="Permanent link">&para;</a></h3>
<p>The concept of a polygenic risk score (PRS) had been circulating among researchers for several years. It wasn’t until 2018, however, that these scores were shown to have potential for broad-scale clinical use. A study by Amit V. Khera and colleagues at the Cardiovascular Disease Initiative of the Broad Institute in Cambridge, Massachusetts, identified people at high risk for five common diseases, based on their genome1.</p>
<p>The team used genome-wide genetic data and imputation methods to assess millions of common genetic variations associated with coronary artery disease, atrial fibrillation, type 2 diabetes, inflammatory bowel disease and breast cancer. For each disease, they applied a computational algorithm that combines information from all of the variants into a number, or PRS, that reflects a person’s inherited susceptibility to these diseases.</p>
<p>When the team tested their polygenic predictor for heart disease on 290,000 participants in the UK Biobank, they found that 8% of the population had 3-times the normal risk of a heart attack. “Those people are totally flying under the radar right now within our clinical practice,” says Khera, who is also a cardiologist and human geneticist at Massachusetts General Hospital. “We can’t find them without doing genetic testing.”</p>
<p>Using the same approach for inflammatory bowel disease and breast cancer, the team found that their PRSs were consistently able to identify a group (between 2-10% of the population) at particularly high risk. Khera's results feed into a growing body of evidence supporting wider use of PRSs.</p>
<p>https://www.nature.com/articles/d42473-019-00270-w</p>
<p>There is risk to broad-scale incorporation of PRS into medicine. Is this 
information to be kept private, or shared? To the extent that a person can 
be easily identified in genealogical studies based on their relatedness to 
other individuals in a genetic database (genetic genealogy research), so too
could someone's PRS be estimated based on their relatedness to other people
for whom a PRS may be known. How might this affect medicine or insurance in 
ways that could be beneficial or harmful?</p>
<h3 id="controversial-gwa-studies">Controversial GWA studies<a class="headerlink" href="#controversial-gwa-studies" title="Permanent link">&para;</a></h3>
<p>We know for a fact that human brain size increased dramatically relative to
our common ancestors. By many measures our brains are also significantly 
rewired in their function relative to other primates as well. Therefore, 
we could extrapolate from this that some mutations must have occurred along
the lineage leading to modern humans that contribute to our development of
larger and more complex brains. (Figure 21.11).</p>
<p>So, if genetic variation can contribute to differences in brain function, 
then does this mean that genetic variation contributes to "intelligence"?
This is question that has been asked for a long time in the study of human
populations, from the application of IQ tests, to eugenic justifications for
the treatment of people from different backgrounds. It is thus a highly 
controversial subject.</p>
<p>We can thus think of many reasons why it may perhaps be better to simply
not ask the question. What can we gain by identifying genetic variants 
associated with <em>some subjective measure of intelligence, such as IQ</em>? 
What is intelligence? An ability to learn X more quickly, or to understand
connections between topics more easily, or to memorize information? How 
can such measures be separated from their cultural context, and from 
education? Do true geniuses exist? When exceptional thinkers or performers, 
such as Einstein, Mozart, or Ramanujan are recognized, is it an outcome of
their genetics? </p>
<p>Even if so, what do we gain by learning this? Should we try to increase the
frequency of these variants in our population? Can we apply genome editing
and/or prenatal screening practices in ways that does not constitute 
immoral eugenics practices? </p>
<p>One argument FOR the study of intelligence, is that this is a complex trait
that is very important to human health, and likely linked to many diseases.
If strong selection on human intelligence fixed many genetic variants in our
history, it may have carried with it many deleterious variants associated
with disease risk. We can better understand these by understanding their link
to brain function, and recent selection. For example, Alzheimers and 
Schizophrenia have been hypothesized to have pleiotropic effects
with intelligence, as have other neurological or pyschiatric disorders.</p>
<hr />







  
  






                
              </article>
            </div>
          
          
<script>var target=document.getElementById(location.hash.slice(1));target&&target.name&&(target.checked=target.name.startsWith("__tabbed_"))</script>
        </div>
        
      </main>
      
        <footer class="md-footer">
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-copyright">
  
  
    Made with
    <a href="https://squidfunk.github.io/mkdocs-material/" target="_blank" rel="noopener">
      Material for MkDocs
    </a>
  
</div>
      
    </div>
  </div>
</footer>
      
    </div>
    <div class="md-dialog" data-md-component="dialog">
      <div class="md-dialog__inner md-typeset"></div>
    </div>
    
    
    <script id="__config" type="application/json">{"base": "..", "features": ["navigation.instant", "navigation.tracking", "search.highlight", "search.suggest", "content.code.annotate"], "search": "../assets/javascripts/workers/search.b8dbb3d2.min.js", "translations": {"clipboard.copied": "Copied to clipboard", "clipboard.copy": "Copy to clipboard", "search.result.more.one": "1 more on this page", "search.result.more.other": "# more on this page", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents", "search.result.placeholder": "Type to start searching", "search.result.term.missing": "Missing", "select.version": "Select version"}}</script>
    
    
      <script src="../assets/javascripts/bundle.bd41221c.min.js"></script>
      
        <script src="../javascripts/extra.js"></script>
      
        <script src="../javascripts/mathjax.js"></script>
      
        <script src="https://polyfill.io/v3/polyfill.min.js?features=es6"></script>
      
        <script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-mml-chtml.js"></script>
      
    
  </body>
</html>