---
title: Phylogenomics
module: 2
session: 11
day2022: W/10/12/2022
day2023: M/10/11/2023
---

# Phylogenomics

----

## Assignments due for this class (W: 10/11/2023)
- Degnan, James H., and Noah A. Rosenberg. 2009. “Gene Tree Discordance, Phylogenetic Inference and the Multispecies Coalescent.” Trends in Ecology & Evolution 24 (6): 332–40. https://doi.org/10.1016/j.tree.2009.01.009.
- Heliconius Genome Consortium, Kanchon K. Dasmahapatra, James R. Walters, Adriana D. Briscoe, John W. Davey, Annabel Whibley, Nicola J. Nadeau, et al. 2012. “Butterfly Genome Reveals Promiscuous Exchange of Mimicry Adaptations among Species.” Nature 487 (7405): 94. https://doi.org/10.1038/nature11041.
- Courseworks quiz
<!-- - Notebook exercise: tree sequences -->
<!-- - Notebook exercise: ABBA-BABA -->
<!-- - Notebook exercise: Fst -->


## Assignments due for next class (M: 10/16/2023)
- Walsh, Jennifer, Shawn M Billerman, Vanya G Rohwer, Bronwyn G Butcher, and Irby J Lovette. 2020. “Genomic and Plumage Variation across the Controversial Baltimore and Bullock’s Oriole Hybrid Zone.” The Auk 137 (4): ukaa044. https://doi.org/10.1093/auk/ukaa044.
- **Evolution 5th edition, Chapter 8** (Geographic diversity)

----

## Outline
- Discuss MSC model
- Notebook exercise
- *Break*
- Discuss Heliconius paper

---- 

## Slides

- [https://docs.google.com/presentation/d/14HKQb9aOMzY3pgdpg8d88YIiL3Oy_JM-pcFohEsxGqU/edit#slide=id.p](https://docs.google.com/presentation/d/14HKQb9aOMzY3pgdpg8d88YIiL3Oy_JM-pcFohEsxGqU/edit#slide=id.p)


## Lecture Notes


### Phylogenetic Inference

Evolutionary relationships are evident in patterns of **homology** -- characters that are shared among taxa and which were inherited from a common ancestor. 

It is important within an evolutionary context to consider whether a shared trait is derived in the group that shares it, versus whether it is ancestral: 
synapomorphy versus plesiomorphy. Only synapomorphies can be used as phylogenetically informative characters.


### Phylogenetic inference methods
Phylogenetic inference generally involves 3 steps:
1. Propose a phylogenetic hypothesis
2. Calculate a score (e.g., likelihood or parsimony score) of the phylogenetic
model given the data.
3. Propose a change to the tree topology and repeat from step 2.

This procedure can be repeated over many iterations to search the space of 
possible trees to try to find the best one. This is a **heuristic approach**, 
meaning that it uses algorithmic shortcuts to try to find the best tree, instead
of trying to calculate the score of every possible tree, since the number of
possible phylogenetic trees is way way way too big.


### Homoplasy
The repeated evolution of a character state. Homoplasy can occur when the rate of evolution of a character is high, or if there are few character states that it can transition between. In either case, if too many transitions occur back and forth between states, it is difficult to infer ancestral states and to accurately 
count the number of changes that have occurred.


### Methods for inferring phylogenies
- Distance
- Parsimony
- Maximum Likelihood, Bayesian


### Using phylogenies
- Inferring divergence times (dates)
- Measuring rates of evolution
- Phylo Gene Association Tests
- Reconstructing Ancestral States
- Phylogenetic Comparative Methods


### Heliconius genome paper
... coming soon

### Multispecies coalescent paper
... coming soon


----

