---
title: Genetic Variation
session: 7
module: 2
day2022: W/9/28/2022
---

# Genetic Variation and Inheritance

Note: Follow-up about writing workshop, questions about writing, and 
submission of thesis statements on courseworks.

----

## Outline
- Lecture: Genetics I
	- Introduction to Ch. 4: Inheritance, Mutation, Recombination.
	- Introduction to Ch. 7: Genetic Drift.
- *Break*
- Exercise: **Review notebooks 0-1**
- Lecture: Introduction to neutral theory

----

## Assigned for next class
- Kern, "Neutral Theory"
- Jensen, "Neutral Theory"

----

## References
- ...
- ...

----

## Lecture slides
**:octicons-browser-16:{ .notebook-thump } [Link to Lecture Slides](../lectures/genetic-variation/)**

----

## Lecture notes

### Evolutionary Modeling
During the Modern synthesis Wright, Fisher, and Haldane developed 
mathematical models that built upon the new understanding of genetic 
(Mendelian) inheritance to predict changes in alleles frequencies in a 
population over time. 

These models are based on a number of important assumptions that describe 
an *idealized population*: a populatin in which allele frequencies will not 
change unless at least one of five evolutionary processes (below) are acting. 

- selection  
- mutation  
- recombination  
- genetic drift  
- migration (pop-structure)  (- we will return to this more later.)  

This unchanging state is called *Hardy Weinberg* equilibrium. Today we will 
focus on Mendel's rules of inheritance, the assumptions of the mathematical
evolutionary models of the Modern Synthesis, and how the five main 
evolutionary processes represent violations of the model assumptions, 
and why they are useful.



### Genetics Terms
First, let's review some (challenging) genetics terms. Thus far, we have used
terms like locus, allele, gene, gene copy, genotype, haplotype, and genome,
but we haven't been very precise in our definitions of them. 
This is in part because some of these terms can be used in multiple ways, 
and also because their meanings have changed over time as they evolved from 
being abstract terms for talking about inherited material, to more precise 
definitions corresponding to physical DNA. 

**Discussion**: What are the limits that define a locus? What is the 
difference between a gene and locus. 

- Gene & Locus: In the loose sense, gene and loci are used interchangeably to refer to 
a region of the genome.
- Gene: In the strict sense, genes have more definitive boundaries, referring
to regions of the genome that code for proteins, and which have defined
transcriptional start and end positions.
- Locus: refer to arbitrary regions of a genome. Their meaning has evolved with
sequencing technologies, first referring to DNA, RNA, or proteins that could
be separated on a gel to be visualized based on differences in their molecular
weight. Later, with DNA sequencing, it often referred to the region that 
was targeted between sequencing primers. More recently, with increasing 
availability of genome data, it is used to refer to a non-recombined block 
of ancestry along a genome.
- Allele: discrete variation at a locus or site.
- Genotype: A combination of alleles at multiple loci (or entire genome).
Usually referring to a diploid genome, whereas a haploid genome will be 
referred to as a haplotype.
- Gene copy: The number of copies of a gene/locus in a population. For a 
population of N=1000 diploid individuals there are 2N gene copies. Many
evolutionary models are focused on how the genetic composition of these 2N
gene copies change over multiple generations as a way of tracking changes in
allele frequencies. One gene copy may leave more descendants (copies) to the
next generation than another (differential fitness) either due to selection 
or chance.

### Measuring genetic diversity
How do we measure genetic diversity in theory and in practice? What do we 
mean by genetic diversity? Diversity at a single site, locus, or across the
whole genome? For whole genomes, should we average across sites, or across
loci or genomic windows? Do we expect genetic diversity to vary spatially
along the genome? Why?

To quantify diversity at a single locus requires us to first define the 
boundaries of the locus. This is usually done with respect to (1) gene or 
exon/intron boundaries; or (2) the positions of primers designed to target
a region; or (3) with respect to estimated recombination break points; or
(4) using an arbitrary window size.

How do we count how many alleles are in a population at a given locus? 


### Mutation
		- What happens when there is a new mutation at a locus, does that create a new allele? (Yes)
	- How do we estimate allele frequencies without sampling loci from every individual in a population?
		- Modeling. 
			- We can estimate the neutral diversity based on Ne and u.
			- With selection: challenging, sometimes we sequence large cohorts.
				- e.g., the frequency of albinism in europeans is 0.0002, or the frequency of B-hemoglobin allele causing sickle cell in central Africa.

- So far, we have assumed that (genetic) variation exists in a population
and discussed how selection can act on phenotypic variation to cause changes in allele frequencies.
	- Chapter 5?: selection on a single locus.
	- Chapter 6?: quantitative genetic models (many loci of small effects).
	- We discussed adaptations. 
		- Often clearly observed in experiments / human impacts / nature
		- But cannot be assumed for every observed feature of organisms:
			- Spandrels, various ways in which features can evolve for non-adaptive reasons.

- One of your recent readings explored the origins of genetic variation 
(mutation and recombination).
	- Let's explore the history of this a bit...
		- A key concept of mutations is that they are *random*.
		- Classic experiment in evolution class, petri dish. Do not arise by need.
		- LTEE
		- Recent paper

	- Key concept of recombination is linkage (correlation).
		- linkage disequilibrium: "relationship is not random."
		- Example, correlation coefficient between alleles.
			- Classic experiment here at Columbia, discovery of chromosomes.
		- Hunt Morgan: In a paper published in Science in 1911, he concluded that (1) some traits were sex-linked, (2) the trait was probably carried on one of the sex chromosomes, and (3) other genes were probably carried on specific chromosomes as well. 


SKIPPED FOR NOW...
- Why recombination?
	- Breaks up associations between loci.
	- Figure from textbook.
	- Figure from textbook.
	- We will return to the topic/effects of recombination several times.

- Hardy Weinberg...

 
- Genetic Drift
	- Definition
	- Random walk analogy
	- Often the first assumption that we violate of the 5. It can provide the
	simplest explanation for evolution (change in allele frequencies).
		- Let's consider study of allele frequencies at a single locus in a single population:
			- selection (of interest)
			- mutation (rare compared to existing variation)
			- recombination (not relevant if we only care about 1 locus)
			- 


- Neutral Theory
